src/satellite/config.ts

```diff
@@ -34,74 +34,88 @@ export const satelliteDefaults: SatelliteOpts = {
   minSnapshotWindow: 40
 }
 
-export interface SatelliteClientOpts {
-  app: string
-  env: string
-  token: string
-  port: number
-  address: string
-  timeout: number
-  pushPeriod: number
-}
-
 export const satelliteClientDefaults = {
+  env: "default",
   timeout: 3000,
-  pushPeriod: 500
+  pushPeriod: 500,
+  insecure: false
 }
 
-export interface SatelliteClientOverrides {
+const baseDomain = "electric-sql.com"
+
+export interface SatelliteClientOpts {
   app: string
-  env: string
+  env?: string
   token: string
+  host: string
   port: number
-  address: string
   timeout?: number
   pushPeriod?: number
+  insecure?: boolean
 }
 
+
 // Config spec
 export interface ElectricConfig {
   app: string
-  env: string
+  env?: string
   token: string
   migrations?: Migration[],
   replication?: {
-    address: string
+    host: string
     port: number
+    insecure: boolean
   }
   debug?: boolean,
 }
 
+const electricConfigDefaults: Partial<ElectricConfig> = {
+  env: "default"
+}
+
+export const addDefaultsToElectricConfig = (config: ElectricConfig) => {
+  const makeHost = () => `${config.env}.${config.app}.${baseDomain}`
+
+  const host = (config.replication?.host) ? config.replication.host : makeHost()
+  const port = (config.replication?.port) ? config.replication.port : 443
+  const insecure = (config.replication?.insecure) ? config.replication.insecure : false
```

Comment:
nitpick:
```suggestion
  const host = config.replication?.host ?? makeHost()
  const port = config.replication?.port ?? 443
  const insecure = config.replication?.insecure ?? false
```

Reply:
cool. Lots of places where we could make better use of this operator :)